src/main/java/com/plotsquared/plothider/PlotSquaredListener.java

```diff
@@ -0,0 +1,104 @@
+ * Copyright (C) IntellectualSites <https://intellectualsites.com>
+ * Copyright (C) IntellectualSites team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.plotsquared.plothider;
+
+import com.google.common.eventbus.Subscribe;
+import com.plotsquared.core.PlotAPI;
+import com.plotsquared.core.events.*;
+import com.plotsquared.core.location.Location;
+import com.plotsquared.core.plot.Plot;
+import com.plotsquared.core.util.task.TaskManager;
+import com.plotsquared.core.util.task.TaskTime;
+import com.sk89q.worldedit.math.BlockVector2;
+import org.bukkit.World;
+import org.bukkit.event.Listener;
+
+public class PlotSquaredListener implements Listener {
+
+    public static final PlotAPI PLOT_API;
+
+    static {
+        PLOT_API = new PlotAPI();
+    }
+
+    public PlotSquaredListener() {
+        PLOT_API.registerListener(this);
+    }
+
+    @Subscribe
+    public void onPlayerPlotDeniedEvent(PlayerPlotDeniedEvent event) {
+        Plot plot = event.getPlot();
+        if (plot.getFlag(HideFlag.class)) {
+            refreshPlot(event.getPlot());
+        }
+    }
+
+    @Subscribe
+    public void onPlayerPlotTrusted(PlayerPlotTrustedEvent event) {
+        Plot plot = event.getPlot();
+        if (plot.getFlag(HideFlag.class)) {
+            refreshPlot(plot);
+        }
+    }
+
+    @Subscribe
+    public void onPlayerPlotHelperEvent(PlayerPlotHelperEvent event) {
+        Plot plot = event.getPlot();
+        if (plot.getFlag(HideFlag.class)) {
+            refreshPlot(plot);
+        }
+    }
+
+    @Subscribe
+    public void onPlotFlagAddEvent(PlotFlagAddEvent event) {
+        Plot plot = event.getPlot();
+        if (event.getFlag() instanceof HideFlag) {
+            // Waits a bit because the flag is added after the event call.
+            TaskManager.runTaskLater(() -> refreshPlot(plot), TaskTime.ticks(1));
+        }
+    }
+
+    @Subscribe
+    public void onPlotFlagRemoveEvent(PlotFlagRemoveEvent event) {
+        Plot plot = event.getPlot();
+        if (event.getFlag() instanceof HideFlag) {
+            // Waits a bit because the flag is removed after the event call.
+            TaskManager.runTaskLater(() -> refreshPlot(plot), TaskTime.ticks(1));
+        }
+    }
+
+    private void refreshPlot(Plot plot) {
+        for (Plot connectedPlot : plot.getConnectedPlots()) {
```

Comment:
This will work *in most cases*, however if you have roads that are larger than a chunk, those road chunks aren't refreshed although they are part of the plot. Maybe it makes sense to just get min and max of all connected plots and refresh all chunks in that region?

Reply:
I don't really understand the concern you pointed out.
> Maybe it makes sense to just get min and max of all connected plots and refresh all chunks in that region?
_Do you mean a global bottom and top point extracted from connected plots to fetch a _cuboid_ shaped region?_

Reply:
> I don't really understand the concern you pointed out.

Example: If you have plots with a 16 blocks width and roads with a 16 blocks (with borders) and you merge two plots, this method will currently refresh two chunks instead of three.

> Do you mean a global bottom and top point extracted from connected plots to fetch a cuboid shaped region?

Yes, while that might cause refreshing chunks that don't need to be refreshed, it's pretty simple and *good enough*.

Reply:
Oh I see, merged road is not a "real plot," thanks for explanations.